### ISSUE TITLE
Support "individual"-specific intercept `alpha` in categorical_logit_glm

### ISSUE THREAD

**Issue by helske** [stan-dev/math#2746]
The newish [categorical_logit_glm](https://mc-stan.org/docs/2_29/functions-reference/categorical-logit-glm.html) function is a great improvement over "manual" softmax regression in terms of efficiency, but I noticed that the current signatures do not allow a case where each "individual" has their own intercept term `alpha`, i.e. it is an NxM matrix (where N is the number of categories and M is the number of individuals). This is supported in other `glm` versions such as `bernoulli_logit_glm`.

#### Current Version:
v4.3.2
